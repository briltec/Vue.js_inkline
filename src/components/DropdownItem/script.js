import DisabledPropertyMixin from 'inkline/mixins/components/properties/DisabledPropertyMixin';
import TabIndexPropertyMixin from 'inkline/mixins/components/properties/TabIndexPropertyMixin';

import AttributesProviderMixin from 'inkline/mixins/components/providers/AttributesProviderMixin';
import ClassesProviderMixin from 'inkline/mixins/components/providers/ClassesProviderMixin';
import EmitProviderMixin from 'inkline/mixins/components/providers/EmitProviderMixin';

export default {
    name: 'IDropdownItem',
    mixins: [
        DisabledPropertyMixin,
        TabIndexPropertyMixin,

        AttributesProviderMixin,
        ClassesProviderMixin,
        EmitProviderMixin
    ],
    props: {
        tag: {
            type: String,
            default: 'div'
        },
        action: {
            type: [String, Number, Boolean],
            default: undefined
        },
        tabindex: {
            type: [Number, String],
            default: -1
        }
    },
    methods: {
        onClick() {
            this.dispatch('IDropdown', 'menu-item-click', [this.action, this]);
        }
    },
    computed: {
        isTag () {
            return this.attributes.to ? 'router-link' : this.attributes.href ? 'a' : this.tag;
        },
    },
    mounted() {
        this.dispatch('IDropdown', 'dropdown-item-mounted', this);
    },
    destroyed() {
        this.dispatch('IDropdown', 'dropdown-item-destroyed', this);
    }
};
