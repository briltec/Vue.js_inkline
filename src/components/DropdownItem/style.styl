@require '../../css/config';
@require '../../css/mixins';
@require '../DropdownMenu/css/_variables';
@require 'css/_sizes';
@require 'css/_variants';

.dropdown-menu > .item
    display: block;
    width: 100%
    clear: both;
    font-weight: --font-weight-normal
    text-align: inherit // For `<button>`s
    white-space: nowrap // prevent links from randomly breaking onto new lines
    background-color: transparent // For `<button>`s
    border: 0


    &.active,
    &:active
        text-decoration: none

    &.disabled,
    &:disabled
        pointer-events: none
        background-color: transparent
        background-image: none

    @extend .dropdown-menu.-light > .item
    @extend .dropdown-menu.-md > .item