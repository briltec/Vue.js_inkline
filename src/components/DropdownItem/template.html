<div class="item"
    :class="classes['*']"
    :aria-disabled="disabled"
    :tabindex="tabindex"
    @click="onClick">
    <slot></slot>
</div>