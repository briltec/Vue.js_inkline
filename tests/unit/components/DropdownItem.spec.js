import {shallowMount} from '@vue/test-utils';
import DropdownItem from '@inkline/inkline/components/DropdownItem';

describe('Components', () => {
    describe('DropdownItem', () => {
        let wrapper;

        beforeEach(() => {
            wrapper = shallowMount(DropdownItem, {
                methods: {
                    mounted: DropdownItem.mounted,
                    destroyed: DropdownItem.destroyed
                }
            });
        });

        it('should be named correctly', () => {
            expect(DropdownItem.name).toEqual('IDropdownItem');
        });

        it('should render correctly', () => {
            expect(wrapper.html()).toMatchSnapshot();
        });

        describe('props', () => {
            describe('action', () => {
                it('should be defined', () => {
                    expect(wrapper.vm).toHaveProperty('action');
                });
            });

            describe('tabindex', () => {
                it('should be defined', () => {
                    expect(wrapper.vm.tabindex).toBeDefined();
                    expect(wrapper.vm.tabindex).toEqual(-1);
                });
            });
        });

        describe('methods', () => {
            describe('onClick()', () => {
                it('should dispatch "menu-item-click" to IDropdown', () => {
                    const spy = jest.spyOn(wrapper.vm, 'dispatch');

                    wrapper.vm.onClick();

                    expect(spy).toHaveBeenCalled();
                    expect(spy).toHaveBeenCalledWith('IDropdown', 'menu-item-click', [wrapper.vm.action, wrapper.vm]);
                });
            });
        });

        describe('mounted()', () => {
            it('should dispatch "dropdown-item-mounted" to IDropdown', () => {
                const spy = jest.spyOn(wrapper.vm, 'dispatch');

                wrapper.vm.mounted();

                expect(spy).toHaveBeenCalled();
                expect(spy).toHaveBeenCalledWith('IDropdown', 'dropdown-item-mounted', wrapper.vm);
            });
        });

        describe('destroyed()', () => {
            it('should dispatch "dropdown-item-destroyed" to IDropdown', () => {
                const spy = jest.spyOn(wrapper.vm, 'dispatch');

                wrapper.vm.destroyed();

                expect(spy).toHaveBeenCalled();
                expect(spy).toHaveBeenCalledWith('IDropdown', 'dropdown-item-destroyed', wrapper.vm);
            });
        });
    });
});
