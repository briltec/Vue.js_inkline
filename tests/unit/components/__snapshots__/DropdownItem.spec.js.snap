// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Components DropdownItem should render correctly 1`] = `
<button tag="button" tabindex="1" class="button">
  <!---->
  <!---->
  <!----></button>
`;
